docs/user-guide/node-selection/index.md

```diff
@@ -198,3 +198,207 @@ must be satisfied for the pod to schedule onto a node.
+remaining un-ignored taints have the indicated effects on the pod. In particular,
+
+* if there is at least one un-ignored taint with effect `NoSchedule` then Kubernetes will not schedule
+the pod onto that node
+* if there is no un-ignored taint with effect `NoSchedule` but there is at least one un-ignored taint with
+effect `PreferNoSchedule` then Kubernetes will *try* to not schedule the pod onto the node
+* if there is at least one un-ignored taint with effect `NoExecute` then the pod will be evicted from
+the node (if it is already running on the node), and will not be
+scheduled onto the node (if it is not yet running on the node).
+
+For example, imagine you taint a node like this
+
+```shell
+kubectl taint nodes node1 key1=value1:NoSchedule
+kubectl taint nodes node1 key1=value1:NoExecute
+kubectl taint nodes node1 key2=value2:NoSchedule
+```
+
+And a pod has two tolerations:
+
+```yaml
+tolerations: 
+   - key: "key1"
+     operator: "Equal"
+     value: "value1"
+     effect: "NoSchedule"
+   - key: "key1"
+     operator: "Equal"
+     value: "value1"
+     effect: "NoExecute"
+```
+
+In this case, the pod will not be able to shedule onto the node (because there is no
+toleration matching the third taint), but it will be able to continue running if it is
+already running on the node when the taint is added (because the third taint is the only
+one of the three that is not tolerated by the pod).
+
+Normally, if a taint with effect `NoExecute` is added to a node, then any pods that do
+not tolerate the taint will be evicted immediately, and any pods that do tolerate the
+taint will never be evicted. However, a toleration with `NoExecute` effect can specify
+an optional `tolerationSeconds` field that dictates how long the pod will stay bound
+to the node after the taint is added. For example,
+
+```yaml
+tolerations: 
+   - key: "key1"
+     operator: "Equal"
+     value: "value1"
+     effect: "NoExecute"
+     tolerationSeconds: 3600
+```
+
+means that if this pod is running and a matching taint is added to the node, then
+the pod will stay bound to the node for 3600 seconds, and then be evicted. If the
+taint is removed before that time, the pod will not be evicted.
+
+### Example use cases
+
+Taints and tolerations are a flexible way to steer pods away from nodes (or evict
+pods that shouldn't be running). A few of the use cases are
+
+* **dedicated nodes**: If you want to dedicate a set of nodes for exclusive use by
+a particular set of users, you can add a taint to those nodes (say, 
+`kubectl taint nodes nodename dedicated=groupName:NoSchedule`) and then add a corresponding
+toleration to their pods (this would be done most easily by writing a custom
+[admission controller](https://kubernetes.io/docs/admin/admission-controllers/)).
+The pods with the tolerations will then be allowed to use the tainted (dedicated) nodes as
+well as any other nodes in the cluster. If you want to dedicated the nodes to them *and*
```

Comment:
Regarding this "...nodes as well as any other nodes in the cluster." I think it would be helpful to specify this somewhere in the beginning to make it more explicit to avoid any confusion related to their placement as in this issue: https://github.com/kubernetes/kubernetes/issues/39818